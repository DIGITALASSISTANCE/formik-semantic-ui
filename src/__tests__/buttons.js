import React from 'react';
import {render, cleanup} from 'react-testing-library';

import {Button, Form} from '../index';

describe('Buttons', () => {
  afterEach(() => {
    cleanup();
  });

  it('default', () => {
    const {container} = render(
      <Form initialValues={{name: ''}}>
        <Button>Basic Button</Button>
      </Form>
    );
    expect(container).toMatchSnapshot();
  });

  it('Button.Submit', () => {
    const {container} = render(
      <Form initialValues={{name: ''}}>
        <Button.Submit>Basic Button</Button.Submit>
      </Form>
    );
    expect(container).toMatchSnapshot();
  });

  it('Button.Reset', () => {
    const {container} = render(
      <Form initialValues={{name: ''}}>
        <Button.Reset>Basic Button</Button.Reset>
      </Form>
    );
    expect(container).toMatchSnapshot();
  });
});
