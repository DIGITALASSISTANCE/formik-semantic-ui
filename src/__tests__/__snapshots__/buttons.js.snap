// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttons Button.Reset 1`] = `
<div>
  <form
    class="ui form"
  >
    <button
      type="button"
      class="ui basic button"
    >
      Basic Button
    </button>
  </form>
</div>
`;

exports[`Buttons Button.Submit 1`] = `
<div>
  <form
    class="ui form"
  >
    <button
      type="submit"
      class="ui primary button"
    >
      Basic Button
    </button>
  </form>
</div>
`;

exports[`Buttons default 1`] = `
<div>
  <form
    class="ui form"
  >
    <button
      type="button"
      class="ui button"
    >
      Basic Button
    </button>
  </form>
</div>
`;
